packages/flutter/lib/src/rendering/binding.dart

```diff
@@ -541,40 +636,103 @@ mixin RendererBinding on BindingBase, ServicesBinding, SchedulerBinding, Gesture
+      renderView.debugLayer?.toStringDeep() ?? 'Layer tree unavailable for $renderView.',
+  ].join('\n\n');
 }
 
-/// Prints a textual representation of the entire layer tree.
+/// Prints a textual representation of the layer trees.
+///
+/// {@macro flutter.rendering.debugDumpRenderTree}
 void debugDumpLayerTree() {
-  debugPrint(RendererBinding.instance.renderView.debugLayer?.toStringDeep());
+  debugPrint(_debugCollectLayerTrees());
 }
 
-/// Prints a textual representation of the entire semantics tree.
-/// This will only work if there is a semantics client attached.
-/// Otherwise, a notice that no semantics are available will be printed.
+String _debugCollectSemanticsTrees(DebugSemanticsDumpOrder childOrder) {
+  if (RendererBinding.instance.renderViews.isEmpty) {
+    return 'No render tree root was added to the binding.';
+  }
+  const String explanation = 'For performance reasons, the framework only generates semantics when asked to do so by the platform.\n'
+      'Usually, platforms only ask for semantics when assistive technologies (like screen readers) are running.\n'
+      'To generate semantics, try turning on an assistive technology (like VoiceOver or TalkBack) on your device.';
+  final List<String> trees = <String>[];
+  bool printedExplanation = false;
+  for (final RenderView renderView in RendererBinding.instance.renderViews) {
+    final String? tree = renderView.debugSemantics?.toStringDeep(childOrder: childOrder);
+    if (tree != null) {
+      trees.add(tree);
+    } else {
+      String message = 'Semantics not generated for $renderView.';
+      if (!printedExplanation) {
+        printedExplanation = true;
+        message = '$message\n$explanation';
+      }
+      trees.add(message);
+    }
+  }
+  return trees.join('\n\n');
+}
+
+/// Prints a textual representation of the semantics trees.
+///
+/// {@macro flutter.rendering.debugDumpRenderTree}
+///
+/// Semantics trees are only constructed when semantics are enabled (see
+/// [SemanticsBinding.semanticsEnabled]). If a semantics tree is not available,
+/// a notice about the missing semantics tree is printed instead.
 ///
 /// The order in which the children of a [SemanticsNode] will be printed is
 /// controlled by the [childOrder] parameter.
 void debugDumpSemanticsTree([DebugSemanticsDumpOrder childOrder = DebugSemanticsDumpOrder.traversalOrder]) {
-  debugPrint(_generateSemanticsTree(childOrder));
+  debugPrint(_debugCollectSemanticsTrees(childOrder));
 }
 
-String _generateSemanticsTree(DebugSemanticsDumpOrder childOrder) {
-  final String? tree = RendererBinding.instance.renderView.debugSemantics?.toStringDeep(childOrder: childOrder);
-  if (tree != null) {
-    return tree;
+/// Prints a textual representation of the [PipelineOwner] tree rooted at
+/// [RendererBinding.rootPipelineOwner].
+void debugDumpPipelineOwnerTree() {
+  // TODO(goderbauer): Turn this tree into a [DiagnosticableTree] for nicer output.
```

Comment:
This is a good idea.  It'll also reduce the code here.

Reply:
Done. Don't know why I didn't do this in the first place. Implementing it correctly took probably less time then it took me to write the "bad" implementation...